app/api/models.py

```diff
@@ -75,6 +75,20 @@ class SubjectsQueryResponse(BaseModel):
     available_pipelines: dict
 
 
+class DatasetQueryResponse(BaseModel):
```

Comment:
Given that this has all fields in common with `SubjectsQueryResponse` except for `subject_data`, to reduce duplication I'd suggest refactoring out the common fields into a standalone class (you could call that one `DatasetQueryResponse` and have `SubjectsQueryResponse` simply inherit from it instead of from `BaseModel` - similar to how we're using `BaseQueryModel` for the queries.